### DISCUSSION TITLE
Recommended way to load obj in a worker?

### DISCUSSION THREAD

**Discussion by geoff-harper** [kaisalmen/WWOBJLoader#76] [Q&A]
Hey Kai, back once again. I'm playing around with loading a scene and an obj in an `OffscreenCanvas` in a worker. I understand that the parallel loader will go off into another thread, fetch/parse and bring it back to the main thread, but I don't think this is quite what I'm looking for. I'm curious if there's a way to just use the regular loader in a worker? Since I'd already be calling it from a worker anyways. My only roadblock so far is that `OBJLoader2` uses `window` in the load method

```ts
const urlFull = new URL(url, window.location.href).href;
let filename = urlFull;
const urlParts = urlFull.split('/');
if (urlParts.length > 2) {
    filename = urlParts[urlParts.length - 1];
    const urlPartsPath = urlParts.slice(0, urlParts.length - 1).join('/') + '/';
    if (urlPartsPath !== undefined) this.path = urlPartsPath;
}
```

Would there be a way to work around this little snippet? Or maybe parallel is actually what I'm looking for?

**Comment by kaisalmen** [comment]
Hi @geoff-harper, you could comment this block an pass the url from the main to the worker containing the whole application. I am assuming you are using a module worker, so you can just import everything you need via import statements (as you would do on main). You could pass urls from the main when initialsing the worker.
Another idea would be to completely get rid of ˋwindowˋ. Idea is to set a baseurl via a setter/global variable, so the window usage is outside ˋOBJLoader2ˋ and it no longer depends on it. I have to check if there are other problems using the whole loader in a worker. Do want to create a PR for it?

**Reply by geoff-harper** [reply]
I think I can do a quick PR that should resolve the issue with no api changes, I'll open it up in a day or two
